.github/workflows/test-rust.yml

```diff
@@ -33,7 +33,10 @@ jobs:
           submodules: recursive
 
       - name: Service shouldn't depends on juniper::ID
-        run:  bash -c '! grep "juniper::ID" -r ./ee/tabby-webserver/src/service'
+        run: bash -c '! grep "juniper::ID" -r ./ee/tabby-webserver/src/service'
+
+      - name: Only service could depends on tabby_db
+        run: bash -c '! grep "tabby_db::" -r ./ee/tabby-webserver/src --exclude-dir ./ee/tabby-webserver/src/service'
```

Comment:
Not sure if this is the best way to do lints like this, let me know if you know a better solution @boxbeam

Reply:
Seems reasonable enough to me

Reply:
endup using https://github.com/ast-grep/ast-grep as lint